pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -317,8 +325,11 @@ func makeHookPod(hook *deployapi.LifecycleHook, deployment *kapi.ReplicationCont
 		return nil, fmt.Errorf("couldn't clone ResourceRequirements: %v", err)
 	}
 
+	podName := namer.GetPodName(deployment.Name, suffix)
+
 	// Assigning to a variable since its address is required
-	maxDeploymentDurationSeconds := deployapi.MaxDeploymentDurationSeconds
+	maxDeploymentDurationSeconds := deployapi.MaxDeploymentDurationSeconds - int64(time.Since(deployerPod.Status.StartTime.Time).Seconds())
+	glog.V(5).Infof("--> DEBUG: %s timeout is %d seconds", podName, maxDeploymentDurationSeconds)
```

Comment:
I don't think this adds much. You should have a unit test for `makeHookPod` that proves we set the timeout correctly.